src/applications/disability-benefits/view-payments/components/view-payments-lists/payments/Payments.jsx

```diff
@@ -26,22 +27,24 @@ const Payments = ({
   textContent,
   alertMessage,
 }) => {
+  const location = useLocation();
+  const navigate = useNavigate();
   const [currentData, setCurrentData] = useState([]);
-  const [currentPage, setCurrentPage] = useState(1);
-  // Using `useRef` here to avoid triggering a rerender whenever these are
-  // updated
-  const totalPages = useRef(0);
-  const paginatedData = useRef([]);
+  const currentPage = new URLSearchParams(location.search).get('page') || 1;
+  const [totalPages, setTotalPages] = useState(0);
 
-  useEffect(() => {
-    paginatedData.current = paginateData(data);
-    setCurrentData(paginatedData.current[currentPage - 1]);
-    totalPages.current = paginatedData.current.length;
-  }, []);
+  useEffect(
+    () => {
+      const paginatedData = paginateData(data);
+      setCurrentData(paginatedData[currentPage - 1]);
+      setTotalPages(paginatedData.length);
+    },
+    [currentPage, data],
+  );
 
   const onPageChange = page => {
-    setCurrentData(paginatedData.current[page - 1]);
-    setCurrentPage(page);
+    const newURL = `${location.pathname}?page=${page}`;
```

Comment:
I'd like to use this custom event to also set focus on the H3 that announces the number of results. I tested it locally and it's working really well and only a few more lines:

```diff
// Line 33
+ const tableHeadingRef = useRef(null);

const onPageChange = page => {
  const newURL = `${location.pathname}?page=${page}`;
+ if (tableHeadingRef) {
+   tableHeadingRef.current.focus();
+ }
  navigate(newURL);
};

<h3
  className="vads-u-font-size--lg vads-u-font-family--serif"
+ ref={tableHeadingRef}
+ tabIndex={-1}
>
```